registry/remote/repository.go

```diff
@@ -198,6 +201,8 @@ func (r *Repository) client() Client {
 
 // blobStore detects the blob store for the given descriptor.
 func (r *Repository) blobStore(desc ocispec.Descriptor) registry.BlobStore {
+	// TODO this seems dubious. Perhaps we actually want to push
+	// a blob with a manifest media type as a regular blob?
```

Comment:
Ah, sorry, this was really a note to self - I hadn't intended to leave it in the PR.

I still do think it's dubious though - some code might be copying blobs without caring about their media type, and if they happen to use Push, then the blob will suddenly turn into a manifest and might refuse to copy because perhaps its component blobs don't exist.

An example of this is the `oras.Copy` function, which just uses `Push` regardless of whether the blob came from the manifest store or the blob store, so AFAICS it can promote a manifest-typed blob to the manifest store which isn't making a true copy of the original. A manifest can be tagged where a blob cannot, so this has some potential to be a security risk.

(aside: I'm fairly sure that it's a bug that `oras.Copy` copies manifests recursively - it means that it follows references that aren't technically live. Here's a reproducer shell script: https://gist.github.com/rogpeppe/162ed0d29d1defb3d067a8d57643ce24)

Reply:
Removed comment.

Reply:
> I'm fairly sure that it's a bug that oras.Copy copies manifests recursively

Yes, it is a known issue. The fundamental problem is that a registry is not a single storage but two storages while `Target` is modelled as a single storage. We can start a discussion on copying stuffs for _multi-storage targets_.

Reply:
> Yes, it is a known issue.

So it is, thanks! For the record, this is https://github.com/oras-project/oras-go/issues/401 I think.